RELEASE/scripts/autoscend/paths/wildfire.ash

```diff
@@ -2,3 +2,316 @@ boolean in_wildfire()
+}
+
+void wildfire_rainbarrel()
+{
+	//collect rainwater from barrel daily
+	if(!in_wildfire())
+	{
+		return;
+	}
+	if(get_property("_wildfireBarrelHarvested").to_boolean())
+	{
+		return;		//already collected today
+	}
+	visit_url("place.php?whichplace=wildfire_camp&action=wildfire_rainbarrel");
+}
+
+int wildfire_water_cost(string target)
+{
+	//return the cost in water to perform watering operations.
+	if(!in_wildfire())
+	{
+		return 0;
+	}
+	if(!($strings[dust,frack,sprinkle,hose] contains target))
+	{
+		abort("an invalid target was passed to int wildfire_water_cost(string target)");
+	}
+	int count = 0;
+	boolean dusted = get_property("wildfireDusted").to_boolean();
+	boolean fracked = get_property("wildfireFracked").to_boolean();
+	boolean sprinkled = get_property("wildfireSprinkled").to_boolean();
+	switch(target)
+	{
+		case "hose":
+			//how much does having cpt hangk send firefighers to hose down an area cost.
+			return 10 + (10 * get_property("_auto_wildfire_hosed_today").to_int());
+		case "dust":
+			if(dusted) return 0;
+			if(fracked) count++;
+			if(sprinkled) count++;
+			break;
+		case "frack":
+			if(fracked) return 0;
+			if(dusted) count++;
+			if(sprinkled) count++;
+			break;
+		case "sprinkle":
+			if(sprinkled) return 0;
+			if(dusted) count++;
+			if(fracked) count++;
+			break;
+	}
+	return 1000+(count*1000);
+}
+
+boolean LX_wildfire_grease_pump()
+{
+	if(!in_wildfire())
+	{
+		return false;
+	}
+	if(get_property("wildfirePumpGreased").to_boolean())
+	{
+		return false;		//already greased
+	}
+	if(item_amount($item[pump grease]) == 0 && npc_price($item[pump grease]) == 0)
+	{
+		abort("We are showing you did not grease the pump & do not have [pump grease] & cannot buy pump grease. Something is wrong. please fix it");
+	}
+	
+	if(item_amount($item[pump grease]) == 0)
+	{
+		//buy the grease
+		pull_meat(npc_price($item[pump grease]));
+		if(my_meat() >= npc_price($item[pump grease]))
+		{
+			buyUpTo(1, $item[pump grease]);
+		}
+		else
+		{
+			if(get_property("lastSecondFloorUnlock").to_int() < my_ascensions())
+			{
+				return false;	//go do other stuff until spookyraven second floor is unlocked
+			}
+			return autoAdv($location[The Haunted Bedroom]);		//get enough meat to grease the pump
+		}
+	}
+	
+	if(item_amount($item[pump grease]) > 0)
+	{
+		//use the grease
+		use(1, $item[pump grease]);
+		if(!get_property("wildfirePumpGreased").to_boolean())
+		{
+			abort("Failed to use [pump grease] or mafia is tracking it incorrectly. please resolve the issue and run me again");
+		}
+		return true;
+	}
+	return false;
+}
+
+boolean LX_wildfire_pump(int target)
+{
+	//use the pump until we reach target water or run low on adv
+	if(!in_wildfire())
+	{
+		return false;
+	}
+	int start_water = my_wildfire_water();
+	while(my_adventures() > 1+auto_advToReserve() && target > my_wildfire_water())
```

Comment:
Will this skip semi rares?

Reply:
yes... my bad there. I had not considered that. I will make an adjustment to account for those.

Reply:
fixed